exporter/containerimage/export.go

```diff
@@ -247,22 +247,26 @@ func (e *imageExporterInstance) Export(ctx context.Context, src *exporter.Source
 		for _, targetName := range targetNames {
 			if e.opt.Images != nil && e.store {
 				tagDone := progress.OneOff(ctx, "naming to "+targetName)
+				imageClientCtx := ctx
+				if e.opts.Epoch != nil {
+					imageClientCtx = epoch.WithSourceDateEpoch(imageClientCtx, e.opts.Epoch)
+				}
 				img := images.Image{
-					Target:    *desc,
-					CreatedAt: time.Now(),
+					Target: *desc,
+					// CreatedAt is ignored (propagated via imageClientCtx)
```

Comment:
Background:
- https://github.com/containerd/containerd/pull/8225

Reply:
Why? You are saying that there is a field `CreatedAt` that `Create()` and `Update()` methods take, but the implementation is broken, and it doesn't work. And instead, only way to pass the value is with this context dance?

Reply:
Yes, but probably s/broken/unsupported by design/.

The `context` was chosen to keep the Go API unmodified.

Reply:
What should we do with this?

Reply:
Is there a containerd issue tracking this? It looks like a bug, and we are trying to work around it.

> keep the Go API unmodified.

I don't see how this even matters in this case. The value is inside a struct, so it doesn't change the function signature, and the value has already been there(and it looks like we were even setting it).

Reply:
> It looks like a bug

Doesn't seem so IIUC: https://github.com/containerd/containerd/pull/8225#issuecomment-1458608838

> it doesn't change the function signature

Still `CreatedAt time.Time` in the struct would have to be changed to `CreatedAt *time.Time` to differentiate "unset" from "1970-01-01 00:00:00".
Alternatively we could add `CreatedAtIsUnset bool` in the struct, but that might look uglier than ctx.

Reply:
> Doesn't seem so IIUC: https://github.com/containerd/containerd/pull/8225#issuecomment-1458608838

I don't think that contradicts. Yes, this is not a build time, but just some time value associated with the containerd image store. But it doesn't change how the value is updated. This is the same value that is returned when querying the `Image` struct.

 > Alternatively we could add CreatedAtIsUnset bool in the struct

I think I would prefer this. It is not a blocker for this PR, but we should have something in containerd that tracks this issue so it doesn't get forgotten.

Reply:
Opened an issue for continuing the discussion:
- https://github.com/containerd/containerd/issues/8322

Reply:
Can we mark more clearly that this is a containerd bug in a comment and needs a follow-up after a fix is done there. There has not been much movement in the containerd issue. I'd like to be sure we are not stuck with this solution forever.

Reply:
Updated comments